newapi/ooniapi/measurements.py

```diff
@@ -1198,6 +1708,175 @@ def coalsum(name):
         return jsonify({"v": 0, "error": str(e)})
 
 
+def _clickhouse_aggregation(
+    resp_format,
+    since,
+    until,
+    inp,
+    domain,
+    category_code,
+    probe_cc,
+    probe_asn,
+    test_name,
+    axis_x,
+    axis_y,
+):
+    log = current_app.logger
+    dimension_cnt = int(bool(axis_x)) + int(bool(axis_y))
+    cacheable = until and until < datetime.now() - timedelta(hours=72)
+    cacheable = False  # FIXME
+
+    # Assemble query
+    colnames = [
+        "anomaly_count",
+        "confirmed_count",
+        "failure_count",
+        "measurement_count",
+    ]
+    cols = [
+        sql.text("countIf(anomaly = 't') AS anomaly_count"),
+        sql.text("countIf(confirmed = 't') AS confirmed_count"),
+        sql.text("countIf(msm_failure = 't') AS failure_count"),
+        sql.text("COUNT(*) AS measurement_count"),
+    ]
+    table = sql.table("fastpath")
+    where = []
+    query_params = {}
+
+    if domain:
+        where.append(sql.text("domain = :domain"))
+        query_params["domain"] = domain
+
+    if inp:
+        where.append(sql.text("input = :input"))
+        query_params["input"] = inp
+
+    if category_code:
+        where.append(sql.text("category_code = :category_code"))
+        query_params["category_code"] = category_code
+        log.error("@"*66)
+
+    if probe_cc:
+        where.append(sql.text("probe_cc = :probe_cc"))
+        query_params["probe_cc"] = probe_cc
+
+    if probe_asn is not None:
+        if probe_asn.startswith("AS"):
```

Comment:
Isn't this already handled earlier on by the `param_asn` function?

Reply:
Updated